### PR TITLE
fix: don't render broken more info links in application

### LINKED ISSUE #1872: Application page has "More info" link on a resource even when the resource does not have "More info" field
E.g.,

image

### PR DESCRIPTION
**Pull request by opqdonut**
... by making urn-catalogue-item-link check that the resid is a urn.

fixes #1872

### CODE REVIEW

File: src/cljc/rems/catalogue_util.cljc [hunk 1]
```diff
@@ -1,9 +1,9 @@
 (ns rems.catalogue-util
   (:require [clojure.string :as str]))
 
-(defn urn-catalogue-item? [{:keys [resid]}]
-  (and resid (str/starts-with? resid "urn:")))
+(defn urn? [resid]
```
Comment (okahilak) [inline src/cljc/rems/catalogue_util.cljc]: nit: This could be private now.

Comment (opqdonut) [reply src/cljc/rems/catalogue_util.cljc]: yeah, but it's a potentially useful util elsewhere too

**Comment by opqdonut** [comment]
guide broke, made `urn?` tolerate nils

File: src/cljc/rems/catalogue_util.cljc (hunk 1 above)
Comment (okahilak) [reply src/cljc/rems/catalogue_util.cljc]: Ok, fine by me to merge.

**Pull request merged**